tests/components/music_assistant/conftest.py

```diff
@@ -24,6 +31,60 @@ def mock_get_server_info() -> Generator[AsyncMock]:
         yield mock_get_server_info
 
 
+@pytest.fixture(name="music_assistant_client")
+async def music_assistant_client_fixture() -> AsyncGenerator[MagicMock]:
+    """Fixture for a Music Assistant client."""
+    with patch(
+        "homeassistant.components.music_assistant.MusicAssistantClient", autospec=True
+    ) as client_class:
+        client = client_class.return_value
+
+        async def connect() -> None:
+            """Mock connect."""
+            await asyncio.sleep(0)
+
+        async def listen(init_ready: asyncio.Event | None) -> None:
+            """Mock listen."""
+            if init_ready is not None:
+                init_ready.set()
+            listen_block = asyncio.Event()
+            await listen_block.wait()
+            pytest.fail("Listen was not cancelled!")
+
+        client.connect = AsyncMock(side_effect=connect)
+        client.start_listening = AsyncMock(side_effect=listen)
+        client.server_info = ServerInfoMessage(
+            server_id=MOCK_SERVER_ID,
+            server_version="0.0.0",
+            schema_version=1,
+            min_supported_schema_version=1,
+            base_url="http://localhost:8095",
+            homeassistant_addon=False,
+            onboard_done=True,
+        )
+        client.connection = MagicMock()
+        client.connection.connected = True
+        client.players = Players(client)
+        client.player_queues = PlayerQueues(client)
+        client.music = Music(client)
+        client.server_url = client.server_info.base_url
+        client.get_media_item_image_url = MagicMock(return_value=None)
+
+        yield client
+
+
+@pytest.fixture(name="integration")
+async def integration_fixture(
+    hass: HomeAssistant, music_assistant_client: MagicMock
+) -> MockConfigEntry:
+    """Set up the Music Assistant integration."""
+    entry = MockConfigEntry(domain=DOMAIN, data={CONF_URL: "http://localhost:8095"})
+    entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+    return entry
```

Comment:
Why do we have 2 ways to add the config entry and setup the integration?

Reply:
The config entry mocks the entire client so we can load fixture data of players.
If you see optimizations to combine them, all good

Reply:
actually scrap that - it was dead code